app/src/main/java/com/android/periodpals/model/alert/AlertModel.kt

```diff
@@ -0,0 +1,55 @@
+package com.android.periodpals.model.alert
+
+/** Interface representing the Alert model. */
+interface AlertModel {
+
+  /**
+   * Adds a new alert.
+   *
+   * @param alert The alert to be added.
+   * @param onSuccess Callback function to be called on successful addition.
+   * @param onFailure Callback function to be called on failure, with the exception as a parameter.
+   */
+  suspend fun addAlert(alert: Alert, onSuccess: () -> Unit, onFailure: (Exception) -> Unit)
+
+  /**
+   * Retrieves an alert by its ID.
+   *
+   * @param idAlert The ID of the alert to be retrieved.
+   * @param onSuccess Callback function to be called on successful retrieval, with the alert as a
+   *   parameter.
+   * @param onFailure Callback function to be called on failure, with the exception as a parameter.
+   */
+  suspend fun getAlert(idAlert: String, onSuccess: (Alert) -> Unit, onFailure: (Exception) -> Unit)
+
+  /**
+   * Retrieves all alerts in data base.
+   *
+   * @param onSuccess Callback function to be called on successful retrieval, with the list of
+   *   alerts as a parameter.
+   * @param onFailure Callback function to be called on failure, with the exception as a parameter.
+   */
+  suspend fun getAlerts(onSuccess: (List<Alert>) -> Unit, onFailure: (Exception) -> Unit)
```

Comment:
I suggest renaming this to something like `getAllAlerts` to avoid confusing it with `getAlert`
```suggestion
  suspend fun getAllAlerts(onSuccess: (List<Alert>) -> Unit, onFailure: (Exception) -> Unit)
```